src/metabase/sync/sync_metadata/fields.clj

```diff
@@ -264,14 +337,23 @@
   ([database :- i/DatabaseInstance, table :- i/TableInstance]
    (sync-util/with-error-handling (format "Error syncing fields for %s" (sync-util/name-for-logging table))
      (let [db-metadata (db-metadata database table)]
+
+       (update! fields-per-table-hist (count db-metadata))
+
        ;; make sure the instances of Field are in-sync
-       (sync-field-instances! table db-metadata (our-metadata table) nil)
+       (time! sync-fields-timer
+              (sync-field-instances! table db-metadata (our-metadata table) nil))
        ;; now that tables are synced and fields created as needed make sure field properties are in sync
-       (update-metadata! table db-metadata nil)))))
+       (time! update-metadata-timer
+              (update-metadata! table db-metadata nil))))))
 
 
 (s/defn sync-fields!
   "Sync the Fields in the Metabase application database for all the Tables in a DATABASE."
   [database :- i/DatabaseInstance]
-  (doseq [table (sync-util/db->sync-tables database)]
-    (sync-fields-for-table! database table)))
+  (time! sync-and-update-fields-for-db-timer
+         (let [tables (sync-util/db->sync-tables database)]
+           (update! tables-per-db-hist (count tables))
```

Comment:
What are the average number of tables per database sync'd

![tables-per-db](https://user-images.githubusercontent.com/215910/34177188-54ae4a12-e4c8-11e7-8304-e8f39ebe3e51.png)